### PR TITLE
[DUOS-2351][risk=no] Pact Testing Framework for Consent-Sam consumer contracts

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2351

First pass at building out our pact test framework implementation. Starting with all Consent usages of Sam APIs.

Future stories/PRs will tackle publishing provider tests once we are comfortable with current expectations.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/java/org/broadinstitute/consent/pact/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Pact Testing
+
+Pact tests can be run with maven using the `pact-tests` profile:
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/pact/README.md]:
```suggestion
Pact tests can be run with maven using the `pact-tests` profile:

```

nit: spacing

File: src/test/java/org/broadinstitute/consent/pact/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+the published pact (contract expectations) between the two.
+
+### Sequence Diagrams
+
+When developing a consumer test, i.e. a unit test that validates a Consent expectation of some other 
+service, we are operating in a `Consumer` context. When developing a provider test, i.e. a unit test
+that validates an external system's expectation of a Consent-provided behavior, we are operating in
+a `Provider` context
+
+#### Consumer Context
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/pact/README.md:35]:
```suggestion
#### Consumer Context

```

nit: spacing

File: src/test/java/org/broadinstitute/consent/pact/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  participant Sam
+  Sam->>Broker: Publishes intended behaviors
+  Broker->>Broker: Validate contracts
+  Consent->>Consent: Generate expected Sam behaviors
+  Consent->>Consent: Unit test expected behaviors
+  Consent->>Broker: Export expectations (contracts) to broker
+  Broker->>Broker: Validate expectations (contracts)
+```
+
+#### Provider Context
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/pact/README.md:50]:
```suggestion
#### Provider Context

```

nit: spacing

**Review by fboulnois — APPROVED** [review]
lgtm 👍

File: src/test/java/org/broadinstitute/consent/pact/sam/SamPactTests.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+  private static final UserStatus USER_STATUS =
+      new UserStatus()
+          .setUserInfo(new UserInfo()
+            .setUserEmail(USER_STATUS_INFO.getUserEmail())
+            .setUserSubjectId(USER_STATUS_INFO.getUserSubjectId()))
+          .setEnabled( new Enabled()
+            .setAllUsersGroup(USER_STATUS_DIAGNOSTICS.getInAllUsersGroup())
+            .setLdap(true)
+            .setGoogle(USER_STATUS_DIAGNOSTICS.getInGoogleProxyGroup()));
```
Comment (connorlbark) [inline src/test/java/org/broadinstitute/consent/pact/sam/SamPactTests.java]: nit: not uniform spacing

**Review by connorlbark — APPROVED** [review]
looks good! i don't fully understand pact tests conceptually, but i will do some research into them to understand.

also, i am not sure if `mvn clean test -Ppact-tests` is working as intended; when that is called, are ONLY pact-tests supposed to run? Right now, all tests are ran with that command, as if I just did `mvn clean test`.

In addition, pact tests are always ran alongside unit tests - `mvn clean test -Dtest="SamPactTests"` runs the pact tests fine (as well as `mvn clean test -Ppact-tests -Dtest="SamPactTests"`). So, I am not sure what the `-Ppact-tests` is supposed to do?

**Comment by rushtong** [comment]
> also, i am not sure if `mvn clean test -Ppact-tests` is working as intended; when that is called, are ONLY pact-tests supposed to run? Right now, all tests are ran with that command, as if I just did `mvn clean test`.

Good catch - it's supposed to run only the pact tests. Will look at that again, it was working at one point.

File: src/test/java/org/broadinstitute/consent/pact/sam/SamPactTests.java (hunk 4 above)
Comment (rushtong) [reply src/test/java/org/broadinstitute/consent/pact/sam/SamPactTests.java]: This is intentional - note that there is a new object being constructed inside `setUserInfo` and `setEnabled`

**Pull request merged**